java/core/lance-jni/src/error.rs

```diff
@@ -15,149 +15,167 @@
 use std::str::Utf8Error;
 
 use arrow_schema::ArrowError;
-use jni::errors::Error as JniError;
+use jni::{errors::Error as JniError, JNIEnv};
+use lance::error::Error as LanceError;
 use serde_json::Error as JsonError;
-use snafu::{Location, Snafu};
 
-/// Java Exception types
-pub enum JavaException {
-    IllegalArgumentException,
-    IOException,
-    RuntimeException,
+pub type JavaResult<T> = std::result::Result<T, JavaError>;
```

Comment:
@eddyxu @QianZhu PTAL, please mainly take a look this class.
The others are mainly moving classes around, and refactor it to use the new error handling approach

Reply:
@LuQQiu can we add tests to capture those errors? Specifically, whether the clear error message is presented.

Reply:
We usually just name this as

```
type Result<T> = std::result::Result<T, Error>`


Struct Error {
...}
```
So, within this crate, it still reads in a style that is consistent with a Rust project.

Reply:
This is to mimic the Python side, turn Result finally to a special end Result